gammapy/datasets/map.py

```diff
@@ -1206,7 +1207,7 @@ def stack(self, other, nan_to_num=True):
 
     def stat_array(self):
         """Statistic function value per bin given the current model parameters."""
-        return cash(n_on=self.counts.data, mu_on=self.npred().data)
+        return self._fit_statistic.stat_array(self.counts.data, self.npred().data)
```

Comment:
Wouldn't the argument of fit_statistics change depending on which stat func is called ?